src/script/lua_api/l_noise.cpp

```diff
@@ -499,6 +511,51 @@ int LuaPcgRandom::l_rand_normal_dist(lua_State *L)
 	return 1;
 }
 
+int LuaPcgRandom::l_get_state(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaPcgRandom *o = checkObject<LuaPcgRandom>(L, 1);
+
+	u64 state, inc;
+	o->m_rnd.getState(state, inc);
+
+	std::stringstream s_state;
+	
+	s_state << std::hex << std::setw(64) << std::setfill('0') << state;
+	s_state << std::hex << std::setw(64) << std::setfill('0') << inc;
+	
+	lua_pushstring(L, s_state.str().c_str());
+	return 1;
+}
+
+int LuaPcgRandom::l_set_state(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaPcgRandom *o = checkObject<LuaPcgRandom>(L, 1);
+	if (!lua_isstring(L, 2)) {
+		errorstream<<"PcgRandom.set_state(): Expected hex strings"<<std::endl;
```

Comment:
This is not necessary. Throwing the Lua error suffices.

Reply:
Good point, copy and paste the dual error log from l_next.